tests/plugins/test_reactor_config.py

```diff
@@ -347,3 +378,539 @@ def test_odcs_config_invalid_default_signing_intent(self, tmpdir):
+                  ssl_certs_dir: /var/certs
+        """, True),
+    ])
+    def test_get_koji_session(self, config, raise_error):
+        tasker, workflow = self.prepare()
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+
+        if raise_error:
+            with pytest.raises(Exception):
+                read_yaml(config, 'schemas/config.json')
+            return
+        else:
+            config_json = read_yaml(config, 'schemas/config.json')
+        workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] = config_json
+
+        auth_info = {
+            "proxyuser": config_json['koji']['auth'].get('proxyuser'),
+            "ssl_certs_dir": config_json['koji']['auth'].get('ssl_certs_dir'),
+            "krb_principal": config_json['koji']['auth'].get('krb_principal'),
+            "krb_keytab": config_json['koji']['auth'].get('krb_keytab_path')
+        }
+
+        (flexmock(atomic_reactor.koji_util)
+            .should_receive('create_koji_session')
+            .with_args(config_json['koji']['hub_url'], auth_info)
+            .once()
+            .and_return(True))
+
+        get_koji_session(workflow)
+
+    @pytest.mark.parametrize(('config', 'raise_error'), [
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  password: testpasswd
+                  username: testuser
+        """, False),
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  ssl_certs_dir: /var/certs
+        """, False),
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  ssl_certs_dir: /var/certs
+                  password: testpasswd
+                  username: testuser
+        """, True),
+
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  ssl_certs_dir: /var/certs
+                  password: testpasswd
+        """, True),
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  ssl_certs_dir: /var/certs
+                  username: testuser
+        """, True),
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  username: testuser
+        """, True),
+
+        ("""\
+          version: 1
+          pulp:
+              name: my-pulp
+              auth:
+                  password: testpasswd
+        """, True),
+    ])
+    def test_get_pulp_session(self, config, raise_error):
+        tasker, workflow = self.prepare()
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+
+        if raise_error:
+            with pytest.raises(Exception):
+                read_yaml(config, 'schemas/config.json')
+            return
+        else:
```

Comment:
Is `else:` needed here? Likewise elsewhere.

Reply:
ahh will delete